SportsHub/SportsHub.Api/Controllers/CommentController.cs

```diff
@@ -1,33 +1,65 @@
 ﻿using AutoMapper;
+using FluentValidation;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using SportsHub.Api.Validations;
+using SportsHub.AppService.Authentication.Models.DTOs;
 using SportsHub.AppService.Services;
+using SportsHub.Domain.Constants;
+using SportsHub.Domain.Models;
 
 namespace SportsHub.Api.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class CommentController : ControllerBase
     {
-        private readonly ICommentService _service;
+        private readonly ICommentService _commentService;
         private readonly IMapper _mapper;
+        private readonly IValidator<PostCommentDTO> _commentValidator;
+        private readonly IGenerateModelStateDictionary _generateModelStateDictionary;
 
-        public CommentController(ICommentService service, IMapper mapper)
+        public CommentController(ICommentService service, IMapper mapper, IValidator<PostCommentDTO> commentValidator, IGenerateModelStateDictionary generateModelStateDictionary)
         {
-            _service = service;
+            _commentService = service;
             _mapper = mapper;
+            _commentValidator = commentValidator;
+            _generateModelStateDictionary = generateModelStateDictionary;
         }
 
         [HttpGet("GetByArticle")]
-        public async Task<IActionResult> GetByArticleAsync(int articleId)
+        public async Task<ActionResult<IEnumerable<Comment>>> GetByArticleAsync(int articleId)
         {
-            var comments = await _service.GetByArticleAsync(articleId);
+            var comments = await _commentService.GetByArticleAsync(articleId);
 
             if (!comments.Any())
             {
-                return BadRequest($"No comments for this article");
+                return BadRequest(ValidationMessages.NoCommentsForArticle);
             }
 
             return Ok(comments);
         }
+
+        [Authorize]
+        [HttpPost("PostComment")]
+        public async Task<ActionResult> PostCommentAsync([FromBody] PostCommentDTO commentInput)
+        {
+            var validationResult = await _commentValidator.ValidateAsync(commentInput);
+
+            if (!validationResult.IsValid)
+            {
+                var response = _generateModelStateDictionary.modelStateDictionary(validationResult);
+                return ValidationProblem(response);
+            }
+
+            var created = await _commentService.PostCommentAsync(commentInput);
+
+            if (!created)
+            {
+                return BadRequest(ValidationMessages.UnableToPostComment);
```

Comment:
Please do not use HTTP verbs in response messages.

Reply:
Fixed.